### ISSUE TITLE
Not reading config file when passing -w

### ISSUE THREAD

**Issue by micah** [extremeshok/clamav-unofficial-sigs#17]
I needed to add a whitelist entry, so i passed -w to enter the signature, and it failed:

```
root@mx1:/# /usr/local/sbin/clamav-unofficial-sigs.sh -c /etc/clamav-unofficial-sigs.conf -w
======================================================================
 eXtremeSHOk.com ClamAV Unofficial Signature Updater
 Version: v4.5.3 (12 August 2015)
 Copyright (c) Adrian Jon Kriel :: admin@extremeshok.com
======================================================================
Input a third-party signature name that you wish to whitelist due to false-positives
and press enter (do not include '.UNOFFICIAL' in the signature name nor add quote
marks to the input string):
SecuriteInfo.com.Spammer.cvent-planner.com
grep: *.ndb: No such file or directory
Signature 'SecuriteInfo.com.Spammer.cvent-planner.com' could not be found.
This script will only create a whitelise entry in my-whitelist.ign2 for ClamAV
'UNOFFICIAL' third-Party signatures as found in the *.ndb databases.
root@mx1:/# 
```

I ran the script with bash -x to see what happened, and I see its not reading the config at all:
```
root@mx1:/# bash -x /usr/local/sbin/clamav-unofficial-sigs.sh -c /etc/clamav-unofficial-sigs.conf -w
+ default_config=/etc/clamav-unofficial-sigs.conf
+ version=4.5.3
+ minimum_required_config_version=50
+ version_date='12 August 2015'
++ basename /usr/local/sbin/clamav-unofficial-sigs.sh
+ output_ver='clamav-unofficial-sigs.sh 4.5.3 (12 August 2015)'
++ basename /usr/local/sbin/clamav-unofficial-sigs.sh
++ basename /etc/clamav-unofficial-sigs.conf
+ usage='
ClamAV Unofficial Signature Databases Update Script - v4.5.3 (12 August 2015)

   Usage: clamav-unofficial-sigs.sh [OPTION] [PATH|FILE]

        -c      Direct script to use a specific configuration file
                e.g.: '\''-c /path/to/clamav-unofficial-sigs.conf'\''.

        -d      Decode a third-party signature either by signature name
                (e.g: Sanesecurity.Junk.15248) or hexadecimal string.
                This flag will '\''NOT'\'' decode image signatures.

        -e      Hexadecimal encode an entire input string that can be
                used in any '\''*.ndb'\'' signature database file.

        -f      Hexadecimal encode a formatted input string containing
                signature spacing fields '\''{}, (), *'\'', without encoding 
                the spacing fields, so that the encoded signature can
                be used in any '\''*.ndb'\'' signature database file.

        -g      GPG verify a specific Sanesecurity database file
                e.g.: '\''-g filename.ext'\'' (do not include file path).

        -h      Display this script'\''s help and usage information.

        -i      Output system and configuration information for
                viewing or possible debugging purposes.

        -m      Make a signature database from an ascii file containing
                data strings, with one data string per line.  Additional
                information is provided when using this flag.

        -r      Remove the clamav-unofficial-sigs script and all of
                its associated files and databases from the system.

        -s      Clamscan integrity test a specific database file
                e.g.: '\''-s filename.ext'\'' (do not include file path).

        -t      If HAM directory scanning is enabled in the script'\''s
                configuration file, then output names of any third-party
                signatures that triggered during the HAM directory scan.

        -v      Output script version and date information.

        -w      Adds a signature whitelist entry in the newer ClamAV IGN2
                format to '\''my-whitelist.ign2'\'' in order to temporarily resolve
                a false-positive issue with a specific third-party signature.
                Script added whitelist entries will automatically be removed
                if the original signature is either modified or removed from
                the third-party signature database.

Alternative to using '\''-c'\'': Place config file in /etc (/etc/clamav-unofficial-sigs.conf)
'
+ echo ======================================================================
======================================================================
+ echo ' eXtremeSHOk.com ClamAV Unofficial Signature Updater'
 eXtremeSHOk.com ClamAV Unofficial Signature Updater
+ echo ' Version: v4.5.3 (12 August 2015)'
 Version: v4.5.3 (12 August 2015)
+ echo ' Copyright (c) Adrian Jon Kriel :: admin@extremeshok.com'
 Copyright (c) Adrian Jon Kriel :: admin@extremeshok.com
+ echo ======================================================================
======================================================================
+ getopts c:defg:himrs:tvw option
+ case $option in
+ conf_file=/etc/clamav-unofficial-sigs.conf
+ getopts c:defg:himrs:tvw option
+ case $option in
+ no_default_config
+ '[' '!' -s /etc/clamav-unofficial-sigs.conf ']'
+ echo 'Input a third-party signature name that you wish to whitelist due to false-positives'
Input a third-party signature name that you wish to whitelist due to false-positives
+ echo 'and press enter (do not include '\''.UNOFFICIAL'\'' in the signature name nor add quote'
and press enter (do not include '.UNOFFICIAL' in the signature name nor add quote
+ echo 'marks to the input string):'
marks to the input string):
+ read input
SecuriteInfo.com.Spammer.cvent-planner.com
+ '[' -n SecuriteInfo.com.Spammer.cvent-planner.com ']'
+ cd ''
++ echo SecuriteInfo.com.Spammer.cvent-planner.com
++ tr -d '"'
++ tr -d \'
+ input=SecuriteInfo.com.Spammer.cvent-planner.com
++ grep -H SecuriteInfo.com.Spammer.cvent-planner.com: '*.ndb'
grep: *.ndb: No such file or directory
+ sig_full=
++ echo ''
++ cut -d : -f2
+ sig_name=
+ '[' -n '' ']'
+ echo 'Signature '\''SecuriteInfo.com.Spammer.cvent-planner.com'\'' could not be found.'
Signature 'SecuriteInfo.com.Spammer.cvent-planner.com' could not be found.
+ echo 'This script will only create a whitelise entry in my-whitelist.ign2 for ClamAV'
This script will only create a whitelise entry in my-whitelist.ign2 for ClamAV
+ echo ''\''UNOFFICIAL'\'' third-Party signatures as found in the *.ndb databases.'
'UNOFFICIAL' third-Party signatures as found in the *.ndb databases.
+ exit
root@mx1:/# 
```

as you can see, the work_dir is not being set, because it is not reading the config file!

**Comment by extremeshok** [comment]
Looks more like a permission error. Have does the script user have read permission of the config and config dir ?

**Comment by extremeshok** [comment]
Fixed in the latest version ( 4.6.0 )
